python_modules/libraries/dagster-dg/dagster_dg/cli/dev.py

```diff
@@ -106,6 +106,20 @@ def dev_command(
         *(["--verbose"] if dg_context.config.cli.verbose else []),
     ]
 
+    import json
+
+    if dg_context.is_workspace:
+        os.environ["DAGSTER_PROJECT_ENV_FILE_PATHS"] = json.dumps(
+            {
+                dg_context.with_root_path(project.path).code_location_name: str(project.path)
+                for project in dg_context.project_specs
+            }
+        )
+    else:
+        os.environ["DAGSTER_PROJECT_ENV_FILE_PATHS"] = json.dumps(
+            {dg_context.code_location_name: str(dg_context.root_path)}
```

Comment:
this is where this back-compat check would go right? Is it worth talking through in more detail what that should look like?

Reply:
Updated to add a new backcompat test - we'll print to the console if any locations have `dagster < 1.10.7`